zerver/views/invite.py

```diff
@@ -39,11 +39,16 @@ def invite_users_backend(
     stream_ids: List[int] = REQ(validator=check_list(check_int)),
 ) -> HttpResponse:
 
-    if (
-        user_profile.realm.invite_to_realm_policy == Realm.INVITE_TO_REALM_ADMINS_ONLY
-        and not user_profile.is_realm_admin
-    ):
-        raise OrganizationAdministratorRequired()
+    if not user_profile.can_invite_others_to_realm():
+        if user_profile.realm.invite_to_realm_policy == Realm.POLICY_ADMINS_ONLY:
+            return json_error(_("Only administrators can invite others to this organization."))
+        if user_profile.realm.invite_to_realm_policy == Realm.POLICY_MODERATORS_ONLY:
+            return json_error(
+                _("Only administrators and moderators can invite others to this organization.")
+            )
+        if user_profile.realm.invite_to_realm_policy == Realm.POLICY_FULL_MEMBERS_ONLY:
+            return json_error(_("Your account is too new to invite others to this organization."))
+        # Guest case will be handled by require_member_or_admin decorator.
```

Comment:
This mostly works, but I feel like this structure of having this block of error-handling checks feels really fragile, and we should figure out how to refactor it.  One thought is that we should use common error messages -- e.g. "Must be an organization administrator" or "Must be a moderator or organization administrator" -- that don't name the specific thing you're trying to do (To simplify translations), though it'd be nice to also encode that we're 

Probably fine to leave this as a future problem, but it does seem important.

That said, I think even for a temporary state, we should have a catch-all clause (even a `raise AssertionError()`) -- here if we added a new policy value, and made `user_profile.can_invite_others_to_realm` return false, this structure would cause that to get through.